### DISCUSSION TITLE
Progress Indicator

### DISCUSSION THREAD

**Discussion by cindyebi** [visual-framework/vf-core#1648] [Proposals]
Following up [the previous discussion](https://github.com/visual-framework/vf-core/discussions/1643) about new components, I thought it's better to separate them into different threads. 

**### VF-Progress-Indicator**
Initially, this component was created to address the need of the Training Academy's personal space project where users could track their learning progress by creating an account. The following mockups show different states and ways of labeling.

Mockups:
v1: word-done progress indicator 
![v2](https://user-images.githubusercontent.com/75834903/131517471-ba14d5c0-639c-427d-9e6b-c110cc0e86d5.png)

v2: percent-done progress indicator
![v1](https://user-images.githubusercontent.com/75834903/131517470-3b8471e8-e722-4e2a-b20b-bad7ba238165.png)

Some thoughts:
- At the moment, the purpose of this component is meant to show progress indicators only (users can't interact with it), not to be confused with a [_progress tracker_](https://uxplanet.org/progress-trackers-in-ux-design-4319cef1c600), though [Carbon Design](https://www.carbondesignsystem.com/components/progress-indicator/usage/) uses the "Progress Indicator" naming convention to show step-by-step progress. I think it will be helpful to clarify the scope of the definition and implementation of this component at this stage. 
- [NNGroup](https://www.nngroup.com/articles/progress-indicators/) specifically called this component a "Percent-done animation progress indicator" that should be used for longer processes that took more than 10seconds.
- [Material Design](https://material.io/components/progress-indicators#usage) defines Progress Indicator similar to a spinner or loader component but visualized differently.

**Comment by cindyebi** [comment]
@khawkins98 @nitinja shall we group the [spinner loader](https://github.com/visual-framework/vf-core/issues/1332) and this progress bar under "vf-progress-indicator"?

**Reply by nitinja** [reply]
I think that the plan already?

**Comment by nitinja** [comment]
started on this component, thanks
